tests/wakunode_rest/test_rest_cors.nim

```diff
@@ -0,0 +1,269 @@
+{.used.}
+
+import
+  stew/shims/net,
+  testutils/unittests,
+  presto,
+  presto/client as presto_client,
+  libp2p/peerinfo,
+  libp2p/multiaddress,
+  libp2p/crypto/crypto
+import
+  ../../waku/waku_node,
+  ../../waku/node/waku_node as waku_node2,
+  ../../waku/waku_api/rest/server,
+  ../../waku/waku_api/rest/client,
+  ../../waku/waku_api/rest/responses,
+  ../../waku/waku_api/rest/debug/handlers as debug_api,
+  ../../waku/waku_api/rest/debug/client as debug_api_client,
+  ../testlib/common,
+  ../testlib/wakucore,
+  ../testlib/wakunode
+
+
+type
+  TestResponseTuple = tuple[status: int, data: string, headers: HttpTable]
+
+proc testWakuNode(): WakuNode =
+  let
+    privkey = crypto.PrivateKey.random(Secp256k1, rng[]).tryGet()
+    bindIp = parseIpAddress("0.0.0.0")
+    extIp = parseIpAddress("127.0.0.1")
+    port = Port(58000)
+
+  newTestWakuNode(privkey, bindIp, port, some(extIp), some(port))
+
+proc fetchWithHeader(request: HttpClientRequestRef): Future[TestResponseTuple]
+    {.async: (raises: [CancelledError, HttpError]).} =
```

Comment:
No need to change because this is a test and I'm happy with that but, I suggest to always return a `Result` and avoid raising any exception